include/pybind11/detail/common.h

```diff
@@ -25,8 +25,10 @@
 // on the main `pybind11` namespace.
 #if !defined(PYBIND11_NAMESPACE)
 #    ifdef __GNUG__
```

Comment:
nit `VISIBILITY` sounds like what is being changed, but this doesn't provide an option to choose.

Consider using `HIDDEN` to make intent more clear.